cpp/src/gandiva/precompiled/extended_math_ops_test.cc

```diff
@@ -93,6 +93,9 @@ TEST(TestExtendedMathOps, TestRoundDecimal) {
   EXPECT_FLOAT_EQ(round_float32_int32(-1234.4567f, 3), -1234.457f);
   EXPECT_FLOAT_EQ(round_float32_int32(-1234.4567f, -3), -1000);
   EXPECT_FLOAT_EQ(round_float32_int32(1234.4567f, 0), 1234);
+  EXPECT_FLOAT_EQ(round_float32_int32(static_cast<float>(1.55), 1), 1.5f);
```

Comment:
Please add an explicit test for round_float32_int32(1.5499999523162842f, 1) = 1.5f

The test here is confusing - it relies on people knowing that (float)1.55 = 1.5499999523162842

Reply:
Done.